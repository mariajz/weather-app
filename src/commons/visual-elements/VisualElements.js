import React from 'react';
import { StyledBackgroundImage, StyledIcon } from './VisualElements.style';
import { icons } from '../../../assets/icons';

export const BackgroundImage = () => {
    return (
        <StyledBackgroundImage
            source={require('../../../assets/images/background.png')}
            blurRadius={100}
            resizeMode="stretch"
            testID="bgImage"
        />
    );
};

export const Theme = {
    BackgroundWhite: opacity => `rgba(255,255,255, ${opacity})`,
};

export const Icon = ({ iconName }) => {
    const path = icons[iconName];
    return <StyledIcon testID="icon" source={path} />;
};
