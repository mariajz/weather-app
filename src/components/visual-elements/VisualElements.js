import React from 'react';
import { StyledBackgroundImage } from './VisualElements.style';

export const BackgroundImage = () => {
    return (
        <StyledBackgroundImage
            source={require('../../../assets/background-2.png')}
            blurRadius={100}
            resizeMode="stretch"
        />
    );
};
