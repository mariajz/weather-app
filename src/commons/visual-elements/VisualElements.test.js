import { render } from '@testing-library/react-native';
import React from 'react';
import { BackgroundImage, Theme, Icon } from './VisualElements';

const { BackgroundWhite } = Theme;

describe('VisualElements', () => {
    it('should render BackgroundImage', () => {
        const container = render(<BackgroundImage />);
        const { getByTestId } = container;

        expect(getByTestId('bgImage').props.source).toStrictEqual({
            testUri: '../../../assets/images/background.png',
        });
    });
    it('should render Theme', () => {
        const container = render(<BackgroundWhite opacity={0.2} />);

        expect(container).toMatchSnapshot();
    });
    it('should render Icon when IconName is passed', () => {
        const container = render(<Icon iconName="search" />);
        const { getByTestId } = container;

        expect(getByTestId('icon').props.source).toStrictEqual({
            testUri: '../../../assets/icons/search.png',
        });
    });
});
