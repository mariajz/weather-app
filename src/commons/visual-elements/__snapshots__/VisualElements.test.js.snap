// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualElements should render Theme 1`] = `"rgba(255,255,255, [object Object])"`;
